hub/server.go

```diff
@@ -183,47 +195,31 @@ func (h *Hub) baseHandler(acmeHosts []string) http.Handler {
 	mainRouter.UseEncodedPath()
 	mainRouter.SkipClean(true)
 
-	// Register /healthz and /metrics (if enabledà in way that doesn't pollute the HTTP logs.
+	// Register /healthz and /metrics (if enabled in way that doesn't pollute the HTTP logs).
 	mainRouter.HandleFunc("/healthz", func(w http.ResponseWriter, r *http.Request) {
 		fmt.Fprint(w, "ok")
 	}).Methods("GET", "HEAD")
 
-	if h.config.GetBool("metrics") {
-		r := mainRouter.PathPrefix("/").Subrouter()
-
-		expectedLogin := h.config.GetString("metrics_login")
-		expectedPassword := h.config.GetString("metrics_password")
-		if expectedLogin != "" && expectedPassword != "" {
-			r.Use(basicAuthMiddleware(expectedLogin, expectedPassword))
-		}
-
-		h.metrics.Register(r)
-	}
-
 	handler := h.chainHandlers(acmeHosts)
 	mainRouter.PathPrefix("/").Handler(handler)
 
 	return mainRouter
 }
 
+func (h *Hub) metricsHandler() http.Handler {
+	router := mux.NewRouter()
+
+	router.HandleFunc("/healthz", func(w http.ResponseWriter, r *http.Request) {
```

Comment:
Maybe could we create a new `registerHealthz`function doing this, to factorize both `/healthz` handlers.

Reply:
Done.